AEPCore/Sources/rules/LaunchRulesEngine.swift

```diff
@@ -283,33 +283,33 @@ public class LaunchRulesEngine {
         let dict = replaceToken(in: consequence.details, data: data)
         return RuleConsequence(id: consequence.id, type: consequence.type, details: dict)
     }
+    
+    private func replaceToken(in value: Any, data: Traversable) -> Any {
+        switch value {
+        case let valString as String:
+            return replaceToken(for: valString, data: data)
+        case let nestedDict as [String: Any?]:
+            return replaceToken(in: nestedDict, data: data)
+        case let nestedArray as [Any]:
+            return replaceToken(in: nestedArray, data: data)
+        default:
+            return value
+        }
+    }
 
     private func replaceToken(in dict: [String: Any?], data: Traversable) -> [String: Any?] {
         var mutableDict = dict
         for (key, value) in mutableDict {
-            switch value {
-            case is String:
-                mutableDict[key] = replaceToken(for: value as! String, data: data)
-            case is [String: Any]:
-                let valueDict = mutableDict[key] as! [String: Any]
-                mutableDict[key] = replaceToken(in: valueDict, data: data)
-            case is [Any]:
-                let valueArr = mutableDict[key] as! [Any]
-                let tokenReplacedArr = valueArr.map({ (val: Any) -> Any in
-                    if val is String {
-                        return replaceToken(for: val as! String, data: data)
-                    } else {
-                        return val
-                    }
-                })
-                mutableDict[key] = tokenReplacedArr
-            default:
-                break
-            }
+            guard let value = value else { break }
```

Comment:
We should not exit the loop when encountering null value. Can you change the logic so that it has same behaviour as before
```
     if (let value = value) {
          mutableDict[key] = replaceToken(in: value, data: data)
      }
```